include/usb/class/usb_cdc.h

```diff
@@ -165,4 +165,26 @@ struct cdc_ecm_descriptor {
 	uint8_t bNumberPowerFilters;
 } __packed;
 
+/**
+ * @typedef cdc_acm_arduino_reset_callback_t
+ * @brief A function that is called when the CDC ACM baud rate is set to
+ * 1200.
+ *
+ * @param port Device struct for the CDC ACM device.
+ */
+typedef void (*cdc_acm_arduino_reset_callback_t)(struct device *port);
+
+/**
+ * @brief Set the callback for remote reset events.
+ *
+ * The callback will be invoked if the remote host sets the virtual baud
+ * rate to 1200 baud.  This is used by Arduino style programmers to reset
+ * the device into the bootloader.
+ *
+ * @param dev       CDC ACM device structure.
+ * @param callback  Event handler.
+ */
+void cdc_acm_arduino_reset_callback_set(
```

Comment:
This needs an `int` return value that can be provide `-ENOSYS`  (or `-ENOTSUP`) if something goes wrong.

Reply:
Done.

Note that I modeled this on the other callback setters which also return void.